Update onpremisespublishing.md (#16556)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/onPremisesApplicationSegment.md

> ---

Reviewer: Change the file name to lowercase characters

title: "onPremisesApplicationSegment resource type"
description: "Represents onPremisesApplicationSegment object when publishing an on-premises wildcard application with Azure AD Application Proxy."
ms.localizationpriority: medium
author: "dhruvinshah"
ms.prod: "applications"
doc_type: "resourcePageType"
---

# onPremisesApplicationSegment resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Represents [onPremisesApplicationSegment](onPremisesApplicationSegment.md) object when publishing an on-premises wildcard application with Azure AD Application Proxy. This resource is used for setting application segment for a particular wildcard application.

Reviewer: Same Q as in the metadata
```suggestion
Represents the configuration for wildcard application publishing through Azure AD Application Proxy. This provides an efficient way to configure many on-premises applications with the same settings. Use this resource to configure the **onPremisesApplicationSegments** property for a particular wildcard application.
```

## Properties

| Property     | Type        | Description |
|:-------------|:------------|:------------|

> |alternateUrl|String|If you are configuring a traffic manager in front of multiple App Proxy application segments, the alternateUrl is the user-friendly URL that will point to the traffic manager.|
> |corsConfigurations|[corsConfiguration](corsConfiguration.md) Collection|CORS Rule definition for particular application segment.|

Reviewer: ```suggestion
|alternateUrl|String|If you are configuring a traffic manager in front of multiple App Proxy application segments, the **alternateUrl** is the user-friendly URL that will point to the traffic manager.|
|corsConfigurations|[corsConfiguration](corsConfiguration.md) Collection|CORS rule definition for particular application segment.|
```

> |externalUrl|String |The published external url for the application segment. For example, https://intranet.contoso.com/|
> |internalUrl|String |The internal url of the application segment. For example, https://intranet/.|

Reviewer: Minor formatting
```suggestion
|externalUrl|String |The published external url for the application segment. For example, `https://intranet.contoso.com/`.|
|internalUrl|String |The internal url of the application segment. For example, `https://intranet/`.|
```

## JSON representation

The following is a JSON representation of the resource.

<!-- {
  "blockType": "resource",
  "optionalProperties": [

  ],
  "@odata.type": "microsoft.graph.onPremisesApplicationSegment",

>   "baseType": null

Reviewer: ```suggestion
```

}-->

```json

> {
> "alternateUrl":"String",
> "corsConfigurations":[{"@odata.type": "microsoft.graph.corsConfiguration"}],
> "externalUrl":"String",
> "internalUrl":"String"
> }

Reviewer: Always include the @odata.type. Copied from the stubs.
```suggestion
{
  "@odata.type": "#microsoft.graph.onPremisesApplicationSegment",
  "externalUrl": "String",
  "internalUrl": "String",
  "alternateUrl": "String",
  "corsConfigurations": [
    {
      "@odata.type": "microsoft.graph.corsConfiguration"
    }
  ]
}
```
